### DISCUSSION TITLE
Unhandled Runtime Error : Hydration failed because the initial UI does not match what was rendered on the server.  Warning: Expected server HTML to contain a matching <div> in <a>.

### DISCUSSION THREAD

**Discussion by dikshyant3** [vercel/next.js#59395] [App Router]
I have encountered with the hydration failed error when I put Link tag before the second framer motion when I remove the Link tag the error disappear how to resolve this error. What can be the steps I can take to resolve this error as I am using NextJS 14

The below is the code snippet.

"use client"

import { motion } from 'framer-motion'
import Link from 'next/link'
import React from 'react'
import { SocialIcon } from 'react-social-icons'

type Props = {}

function Header({ }: Props) {
    return (
        <header className='sticky top-0 p-4 flex items-start justify-between max-w-7xl mx-auto z-20 xl:items-center'>
            <motion.div initial={{ x: -500, opacity: 0, scale: 0.5 }} animate={{ x: 0, opacity: 1, scale: 1 }} transition={{ duration: 1.5, }} className='flex flex-row items-center'>
                {/* Social Icons */}
                <SocialIcon url="https://twitter.com/dikdhungana2" fgColor='gray' bgColor='transparent' target='_blank' />
                <SocialIcon url="https://www.linkedin.com/in/dikshyant-dhungana-22b7331a6/" fgColor='gray' bgColor='transparent' target='_blank' />
                <SocialIcon url="https://github.com/dikshyant3" fgColor='gray' bgColor='transparent' target='_blank' />
            </motion.div>

            <Link href="contact">
                <motion.div initial={{ x: 500, opacity: 0, scale: 0.5 }} animate={{ x: 0, opacity:     1, scale: 1 }} transition={{ duration: 1.5, }} className='flex flex-row items-center text-gray-300 cursor-pointer'>
                    {/* Email */}
                    <SocialIcon className="cursor-pointer" network='email'
                        fgColor='gray' bgColor='transparent' />
                    <p className='uppercase hidden md:inline-flex text-sm text-gray-400'>Get in Touch</p>

                </motion.div>
            </Link>
        </header>
    )
}

export default Header

The below is the package.json

{
  "name": "porfolio",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@heroicons/react": "^2.0.18",
    "@react-email/components": "0.0.12",
    "@react-email/tailwind": "0.0.13",
    "class-variance-authority": "^0.7.0",
    "clsx": "^2.0.0",
    "framer-motion": "^10.16.12",
    "lucide-react": "^0.294.0",
    "next": "^14.0.3",
    "react": "^18",
    "react-dom": "^18",
    "react-simple-typewriter": "^5.0.1",
    "react-social-icons": "^6.7.0",
    "resend": "^2.0.0",
    "tailwind-merge": "^2.1.0",
    "tailwindcss-animate": "^1.0.7"
  },
  "devDependencies": {
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "autoprefixer": "^10.0.1",
    "eslint": "^8",
    "eslint-config-next": "14.0.3",
    "postcss": "^8",
    "tailwind-scrollbar": "^3.0.5",
    "tailwindcss": "^3.3.0",
    "typescript": "^5"
  }
}

I just want to navigate to contact page when I am clicking the icon and text div but it does so by showing errors in the webpage

**Comment by Austin1serb** [comment]
having same issue, cannot figure it out, same error if I just use an <a> tag also

**Comment by icyJoseph** [comment]
Hi,

This most likely because, you have bad HTML, such as nesting p's, nested div's within p's and such... Could any of you try to share the HTML structure and its surroundings? so that we can validate and point at the badly shaped HTML?

**Reply by Austin1serb** [reply]
Yes I figured it out. In the component composition, I had a parent component, with a link tag, and in the child I had a link tag. Therefore a nested link
